src/query/frontend/ast/cypher_main_visitor.cpp

```diff
@@ -1305,27 +1302,73 @@ antlrcpp::Any CypherMainVisitor::visitDenyPrivilege(MemgraphCypher::DenyPrivileg
   AuthQuery *auth = storage_->Create<AuthQuery>();
   auth->action_ = AuthQuery::Action::DENY_PRIVILEGE;
   auth->user_or_role_ = std::any_cast<std::string>(ctx->userOrRole->accept(this));
-  if (ctx->privilegeList()) {
-    for (auto *privilege : ctx->privilegeList()->privilege()) {
-      extractPrivilege(auth, privilege);
-    }
+  if (ctx->privilegesList()) {
+    std::tie(auth->label_privileges_, auth->edge_type_privileges_, auth->privileges_) =
+        std::any_cast<std::tuple<std::vector<std::unordered_map<AuthQuery::LabelPrivilege, std::vector<std::string>>>,
+                                 std::vector<std::unordered_map<AuthQuery::LabelPrivilege, std::vector<std::string>>>,
+                                 std::vector<memgraph::query::AuthQuery::Privilege>>>(
+            ctx->privilegesList()->accept(this));
   } else {
     /* deny all privileges */
     auth->privileges_ = kPrivilegesAll;
   }
   return auth;
 }
 
+/**
+ * @return std::tuple<std::vector<std::unordered_map<AuthQuery::LabelPrivilege, std::vector<std::string>>>,
+                    std::vector<std::unordered_map<AuthQuery::LabelPrivilege, std::vector<std::string>>>,
+                    std::vector<memgraph::query::AuthQuery::Privilege>>
+ */
+antlrcpp::Any CypherMainVisitor::visitPrivilegesList(MemgraphCypher::PrivilegesListContext *ctx) {
+  std::vector<std::unordered_map<AuthQuery::LabelPrivilege, std::vector<std::string>>> label_privileges{};
+  std::vector<std::unordered_map<AuthQuery::LabelPrivilege, std::vector<std::string>>> edge_type_privileges{};
+  std::vector<memgraph::query::AuthQuery::Privilege> privileges{};
```

Comment:
Also I think it is totally ok to just create alias for these monstrosities, e.g.:
`using PrivilegesList = std::vector<std::unordered_map<AuthQuery::LabelPrivilege, std::vector<std::string>>>;`
```suggestion
  std::vector<std::unordered_map<AuthQuery::LabelPrivilege, std::vector<std::string>>> label_privileges;
  std::vector<std::unordered_map<AuthQuery::LabelPrivilege, std::vector<std::string>>> edge_type_privileges;
  std::vector<memgraph::query::AuthQuery::Privilege> privileges;
```

Reply:
Can we leave aliases for subsequent PR? This is not so easy to implement, because the source of this alias should technically be in generated header

Reply:
Yes, it is just a suggestion, if you feel it is not worth it, just ignore it